src/nvim/change.c

```diff
@@ -154,7 +154,9 @@ void changed(buf_T *buf)
 /// Also used for recovery.
 void changed_internal(buf_T *buf)
 {
-  buf->b_changed = true;
+  if (!bt_prompt(buf)) {
+    buf->b_changed = true;
```

Comment:
I have utterly no clue why this causes an unrelated test fail. It fails on this screen assert

https://github.com/neovim/neovim/blob/d9465e984b34ab7f98007e52edaf1ebc0ccef038/test/functional/legacy/prompt_buffer_spec.lua#L511-L521

The thing is it's not even an assert that's testing anything unique. It's just after `<cr>` / prompt-submit. And I can't reproduce the test failure on my local system. Only happening on CI.

Any clue why this might be happening? Or what can be done to fix it

Reply:
One does not merely set a buffer field :) Looking at assignments to `buf->b_changed`, I see things like this: https://github.com/neovim/neovim/blob/815916b45005da04838ceee749cec875ddad0f1a/src/nvim/change.c#L155-L158

That suggests that `changed_internal` may be the proper way to set bu_changed.